### PR TITLE
Use instance variables for callback instead of class variables

### PR DESCRIPTION
**Pull request by grm**

### CODE REVIEW

File: src/gardena/devices/base_device.py [hunk 1]
```diff
@@ -4,21 +4,20 @@
...
-    battery_state = "N/A"
-    name = "N/A"
-    rf_link_level = "N/A"
-    rf_link_state = "N/A"
-    serial = "N/A"
-    model_type = "N/A"
-    callbacks = []
-
-    def __init__(self, smart_system, device_map):
+    def __init__(self, smart_system, id):
```
Comment (grm) [inline src/gardena/devices/base_device.py:7]: Codacy Issue found: [Missing docstring in __init__ (D107)](https://app.codacy.com/gh/py-smart-gardena/py-smart-gardena/pullRequest?prid=5620922)

**Comment by grm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/py-smart-gardena/py-smart-gardena/pullRequest?prid=5620922&bid=18413754)

File: src/gardena/devices/base_device.py (hunk 1 above)
Comment (grm) [inline src/gardena/devices/base_device.py:7]: Codacy Issue found: [Redefining built-in 'id'](https://app.codacy.com/gh/py-smart-gardena/py-smart-gardena/pullRequest?prid=5620922)

**Comment by grm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/py-smart-gardena/py-smart-gardena/pullRequest?prid=5620922&bid=18413754)

File: src/gardena/devices/smart_irrigation_control.py [hunk 2]
```diff
@@ -4,14 +4,15 @@
...
 class SmartIrrigationControl(BaseDevice):
 
-    valve_set_id = "N/A"
-    valve_set_state = "N/A"
-    valve_set_last_error_code = "N/A"
-    valves = {}
-
     def __init__(self, smart_system, device_map):
-        BaseDevice.__init__(self, smart_system, device_map)
+        """Constructor for the smart irrigation control device"""
```
Comment (grm) [inline src/gardena/devices/smart_irrigation_control.py:8]: Codacy Issue found: [First line should end with a period, question mark, or exclamation point (not 'e') (D415)](https://app.codacy.com/gh/py-smart-gardena/py-smart-gardena/pullRequest?prid=5620922)

File: src/gardena/devices/sensor.py [hunk 3]
```diff
@@ -3,14 +3,15 @@
...
 class Sensor(BaseDevice):
 
-    ambient_temperature = "N/A"
-    light_intensity = "N/A"
-    soil_humidity = "N/A"
-    soil_temperature = "N/A"
-
     def __init__(self, smart_system, device_map):
-        BaseDevice.__init__(self, smart_system, device_map)
+        """Constructor for the sensor device"""
```
Comment (grm) [inline src/gardena/devices/sensor.py:7]: Codacy Issue found: [First line should end with a period, question mark, or exclamation point (not 'e') (D415)](https://app.codacy.com/gh/py-smart-gardena/py-smart-gardena/pullRequest?prid=5620922)

File: src/gardena/devices/base_device.py [hunk 4]
```diff
@@ -4,21 +4,21 @@
...
-    name = "N/A"
-    rf_link_level = "N/A"
-    rf_link_state = "N/A"
-    serial = "N/A"
-    model_type = "N/A"
-    callbacks = []
-
-    def __init__(self, smart_system, device_map):
+    def __init__(self, smart_system, device_id):
+        """Constructor for the BaseDevice"""
```
Comment (grm) [inline src/gardena/devices/base_device.py:8]: Codacy Issue found: [First line should end with a period, question mark, or exclamation point (not 'e') (D415)](https://app.codacy.com/gh/py-smart-gardena/py-smart-gardena/pullRequest?prid=5620922)

File: src/gardena/devices/power_socket.py [hunk 5]
```diff
@@ -4,12 +4,13 @@
 
 class PowerSocket(BaseDevice):
 
-    activity = "N/A"
-    state = "N/A"
-
     def __init__(self, smart_system, device_map):
-        BaseDevice.__init__(self, smart_system, device_map)
+        """Constructor for the Power socket device"""
```
Comment (grm) [inline src/gardena/devices/power_socket.py:8]: Codacy Issue found: [First line should end with a period, question mark, or exclamation point (not 'e') (D415)](https://app.codacy.com/gh/py-smart-gardena/py-smart-gardena/pullRequest?prid=5620922)

File: src/gardena/devices/water_control.py [hunk 6]
```diff
@@ -4,15 +4,16 @@
...
 
-    valve_set_id = "N/A"
-    valve_id = "N/A"
-    valve_activity = "N/A"
-    valve_name = "N/A"
-    valve_state = "N/A"
-
     def __init__(self, smart_system, device_map):
-        BaseDevice.__init__(self, smart_system, device_map)
+        """Constructor for the water control device"""
```
Comment (grm) [inline src/gardena/devices/water_control.py:8]: Codacy Issue found: [First line should end with a period, question mark, or exclamation point (not 'e') (D415)](https://app.codacy.com/gh/py-smart-gardena/py-smart-gardena/pullRequest?prid=5620922)

File: src/gardena/devices/mower.py [hunk 7]
```diff
@@ -4,14 +4,15 @@
...
 class Mower(BaseDevice):
 
-    activity = "N/A"
-    operating_hours = "N/A"
-    state = "N/A"
-    last_error_code = "N/A"
-
     def __init__(self, smart_system, device_map):
-        BaseDevice.__init__(self, smart_system, device_map)
+        """Constructor for the mower device"""
```
Comment (grm) [inline src/gardena/devices/mower.py:8]: Codacy Issue found: [First line should end with a period, question mark, or exclamation point (not 'e') (D415)](https://app.codacy.com/gh/py-smart-gardena/py-smart-gardena/pullRequest?prid=5620922)

**Comment by grm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/py-smart-gardena/py-smart-gardena/pullRequest?prid=5620922&bid=18413754)

File: src/gardena/devices/base_device.py [hunk 8]
```diff
@@ -2,23 +2,23 @@
...
-    rf_link_state = "N/A"
-    serial = "N/A"
-    model_type = "N/A"
-    callbacks = []
-
-    def __init__(self, smart_system, device_map):
+    """Base class informations about gardena devices."""
+
+    def __init__(self, smart_system, device_id):
+        """Constructor for the BaseDevice"""
```
Comment (grm) [inline src/gardena/devices/base_device.py:8]: Codacy Issue found: [First line should end with a period, question mark, or exclamation point (not 'e') (D415)](https://app.codacy.com/gh/py-smart-gardena/py-smart-gardena/pullRequest?prid=5620922)

**Comment by grm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/py-smart-gardena/py-smart-gardena/pullRequest?prid=5620922&bid=18413754)

File: src/gardena/devices/device_factory.py [hunk 9]
```diff
@@ -8,6 +8,7 @@
 class DeviceFactory:
     @staticmethod
     def build(smart_system, device_map):
+        print(f'{device_map}')
```
Comment (osks) [inline src/gardena/devices/device_factory.py:11]: Perhaps remove that debug print?

**Comment by grm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/py-smart-gardena/py-smart-gardena/pullRequest?prid=5620922&bid=18413754)

**Pull request merged**